### PR TITLE
fixes for version guards

### PR DESCRIPTION
**Pull request by elrond79**
- fix version comparisons
  previously, we were comparing with, ie:
  ```
  #if (PXR_MAJOR_VERSION > 0) || (PXR_MINOR_VERSION >= 19 && PXR_PATCH_VERSION >= 5)
  ```
  The intent here was `if version >= 19.05`... but this wasn't what it did -
  for instance, 20.01 would not evaluate `true`.  The correct expression would
  be:
  ```
  #if (PXR_MAJOR_VERSION > 0) || (PXR_MAJOR_VERSION == 0 && (PXR_MINOR_VERSION > 19 || (PXR_MINOR_VERSION == 19 && PXR_PATCH_VERSION >= 5)))
  ```
  However, that gets pretty wordy and hard to read, so I instead defined a
  single numerical constant, USD_VERSION_NUM, and we compare against that:
  ```
  #if USD_VERSION_NUM >= 1905
  ```
  (Note that for 19.11+, we can compare against the `PXR_VERSION` in pxr.h,
  since it's been fixed to be a valid numerical constant.  However, that's
  doesn't help us much, since we need to build against 19.01)
- apply `GlfContextCaps::InitInstance()` fix in 19.03
  This fix was originally included in the pr/usd-0.19.5-update branch, but it
  can actually be applied in 19.03.  I think it only started creating errors
  in 19.05, but it's possible it was simply mis-categorized. In any case,
  including these changes in 19.03 should work, and is probably "best".
- fix `SetCameraState` change so it only applies to 19.11 (ie, > 19.07)

### CODE REVIEW

File: plugin/al/lib/AL_USDMaya/AL/usdmaya/PluginRegister.h [hunk 1]
```diff
@@ -51,7 +51,7 @@
 #include "pxr/base/plug/plugin.h"
 #include "pxr/base/plug/registry.h"
 
-#if (PXR_MAJOR_VERSION > 0) || (PXR_MINOR_VERSION >= 19 && PXR_PATCH_VERSION >= 5) 
+#if USD_VERSION_NUM >= 1903
```
Comment (mattyjams) [inline plugin/al/lib/AL_USDMaya/AL/usdmaya/PluginRegister.h]: Should the version number be 1905 here, based on the incantation being replaced?

File: plugin/al/lib/AL_USDMaya/AL/usdmaya/PluginRegister.h [hunk 2]
```diff
@@ -171,7 +171,7 @@ MStatus registerPlugin(AFnPlugin& plugin)
 {
   GlfGlewInit();
 
-  #if (PXR_MAJOR_VERSION > 0) || (PXR_MINOR_VERSION >= 19 && PXR_PATCH_VERSION >= 5) 
+  #if USD_VERSION_NUM >= 1903
```
Comment (mattyjams) [inline plugin/al/lib/AL_USDMaya/AL/usdmaya/PluginRegister.h]: Similar here, 1905?

**Comment by mattyjams** [comment]
Actually, a question about this now that I think about it some more...

Could we be using PXR_VERSION here instead and avoid adding USD_VERSION_NUM in the maya-usd CMake? Since we're not going to be building maya-usd with CMake, we'd have to inject this value some other way. But it looks like PXR_VERSION is the same thing and is already provided by the USD installation?

File: plugin/al/lib/AL_USDMaya/AL/usdmaya/PluginRegister.h (hunk 2 above)
Comment (elrond79) [reply plugin/al/lib/AL_USDMaya/AL/usdmaya/PluginRegister.h]: This fix was originally included in the pr/usd-0.19.5-update branch, but it
can actually be applied in 19.03. I think it only started creating errors
in 19.05, but it's possible it was simply mis-categorized. In any case,
including these changes in 19.03 should work, and is probably "best".

**Comment by elrond79** [comment]
for 19.11+, we can compare against the PXR_VERSION in pxr.h,
since it's been fixed to be a valid numerical constant. However, that's
doesn't help us much, since we need to build against 19.01.

Prior to 19.11, it's defined as "001911", which isn't a valid numerical constant in C++ (a leading zero marks it as octal - which is bad to begin with - and the 9 is then an invalid digit for octal).

**Pull request merged**